client/routes.js

```diff
@@ -21,6 +21,8 @@ class Routes extends Component {
         {/* Routes placed here are available to all visitors */}
         <Route path="/login" component={Login} />
         <Route path="/signup" component={Signup} />
+        <Route exact path="/products"component={AllProducts} />
+        <Route path="/products/:id"component={SingleProduct} />
```

Comment:
let's give an extra space between `path="/products/:id"` and `component={SingleProduct}`